code/modules/events/event_dynamic.dm

```diff
@@ -32,57 +32,14 @@ var/list/event_last_fired = list()
 
 	//It is this coder's thought that weighting events on job counts is dumb and predictable as hell. 10 Engies ? Hope you like Meteors
 	//Instead, weighting goes from 100 (boring and common) to 10 (exceptional)
+	for(var/type in subtypesof(/datum/event))
+		if((map.event_blacklist.len && map.event_blacklist.Find(type)) || (map.event_whitelist.len && !map.event_whitelist.Find(type)))
+			possibleEvents[type] = 0
+			continue
+		var/datum/event/E = new type(FALSE)
+		possibleEvents[type] = E.can_start(active_with_role)
+		qdel(E)
```

Comment:
I really don't like spawning something just to run one proc and delete it, but I can't think of another way to do it since the `initial()` trick only works with vars